Kernel/Process.h

```diff
@@ -880,7 +882,14 @@ class Process final
 
     LockWeakPtr<Memory::Region> m_master_tls_region;
 
-    IntrusiveListNode<Process> m_jail_list_node;
+    IntrusiveListNode<Process> m_scoped_list_node;
```

Comment:
Do we intend to use this list for anything other than jails?

I think it's confusing to have a "list node" and a "scoped list node". 

The member names describe the type of the node, not what it's used for. 

Same with the list vs scoped list typedefs. What makes it a "scoped list"?

Can we call the members something like "m_all_processes_list_node" and "m_jail_list_node" to make their name match their use case?